detox/src/devices/common/drivers/android/tools/ApkValidator.js

```diff
@@ -0,0 +1,52 @@
+const DetoxRuntimeError = require('../../../../../errors/DetoxRuntimeError');
+
+const setupGuideHint = 'For further assistance, visit the Android setup guide: https://github.com/wix/Detox/blob/master/docs/Introduction.Android.md';
+
+class ApkValidator {
+  constructor(aapt) {
+    this._aapt = aapt;
+  }
+
+  async validateAppApk(binaryPath) {
+    let isAppApk;
+
+    try {
+      isAppApk = !await this._aapt.isTestAPK(binaryPath);
+    } catch (e) {
+      throw this._composeAaptBasedError(e, binaryPath, 'binary');
+    }
```

Comment:
not tested

Reply:
It's covered by "should throw a specific error if AAPT throws" (https://github.com/wix/Detox/pull/3111/files/558eba81d47d302aa4e258910f267343497f8711..42c7aed553a47cff000f05f6bf1093735fb51d9c#diff-e41e8194f7736d596a564bc5396f164b42ff84d5c6e7893f9071042733b5c7f4R33)

Also note that we force 100% coverage so without coverage here we'd have red builds

Reply:
Cool, I missed this one 👍🏼